### PR TITLE
Bump transformers pin

### PR DESCRIPTION
**Pull request by anijain2305**
Trying to update hf pin.

Benchmarking run to figure out issues

image

Retrying - https://github.com/pytorch/pytorch/pull/156118

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @Lucaskabela

### CODE REVIEW

File: benchmarks/dynamo/check_accuracy.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "detectron2_maskrcnn_r_101_c4",
     "timm_efficientnet",  # see https://github.com/pytorch/pytorch/issues/148699
     "XGLMForCausalLM",  # discovered in https://github.com/pytorch/pytorch/pull/128148
+    "moondream",  # discovered in https://github.com/pytorch/pytorch/pull/159291
```
Comment (BoyuanFeng) [inline benchmarks/dynamo/check_accuracy.py:17]: should we file an issue?

File: benchmarks/dynamo/torchbench.py [hunk 2]
```diff
@@ -382,6 +382,16 @@ def load_model(
...
             import torch_xla  # noqa: F401
+
+        # Turning off kv cache for torchbench models. This is not the right
+        # thing to do, but the torchbench models are way outdated, and since we
+        # are using torchbench pt2 dashboard to track regressions (rather than
+        # improving performance), we are just setting the kv cache to false.
+        # Real transformers benchmarks will be added soon using a different
+        # infra.
+        if model_name.startswith("hf") and hasattr(model.config, "use_cache"):
+            model.config.use_cache = False
```
Comment (BoyuanFeng) [inline benchmarks/dynamo/torchbench.py:397]: what would be the issue if we don't turn off kv cache? We probably want to track a real-world setting for regression..

File: .github/ci_commit_pins/torchao.txt [hunk 3]
```diff
@@ -1 +1 @@
-51c87b6ead6b7e098ada95d6a7609ee873b854cf
+6501fb897bce1e8e4160d6131fba214b3519a048
```
Comment (BoyuanFeng) [inline .github/ci_commit_pins/torchao.txt]: I guess the new transformer pin depends on updated torchao pin?

**Review by BoyuanFeng — APPROVED** [review]

**Comment by BoyuanFeng** [comment]
Thanks for bumping the transfomers pin! Should we add screenshot of dashboard before/after this pr to the description? So we know the dashboard change from pin update.

**Comment by anijain2305** [comment]
Converting to draft because the dashboard is red even though CI passes.

File: benchmarks/dynamo/torchbench.py (hunk 2 above)
Comment (anijain2305) [reply benchmarks/dynamo/torchbench.py:397]: We should just rely on Angela's transformers models here, instead of trying to work with these old models. There are a few different types of caches in transfomers DynamicCache, StaticCache etc, and its better to rely on the `generate` API (which Angela uses in the new model benchmarks).

My other extreme suggestion is to just remove these models from the CI, but thats a discussion for another day.

**Review by huydhn — APPROVED** [review]

File: .ci/pytorch/common_utils.sh [hunk 4]
```diff
@@ -283,6 +283,11 @@ function clone_pytorch_xla() {
   fi
 }
 
+function install_transformers() {
+  pip install transformers==4.54.0
```
Comment (huydhn) [inline .ci/pytorch/common_utils.sh]: Just a note, the latest is 4.55 now that supports the fresh-off-the-press `gpt-oss`

**Comment by huydhn** [comment]
(accidentally delete the branch, but I'm re-opening the PR now)

**Comment by huydhn** [comment]
The dashboard run is almost done.  I will update the rest of the accuracy values tomorrow.  Among them, https://github.com/pytorch/pytorch/actions/runs/16821307931/job/47652611212 looks like a real regression but it comes from my change, so I will also push a fix for this in one go.

File: .ci/docker/ci_commit_pins/huggingface.txt [hunk 5]
```diff
@@ -1 +1 @@
-243e186efbf7fb93328dd6b34927a4e8c8f24395
+v4.54.0
```
Comment (huydhn) [inline .ci/docker/ci_commit_pins/huggingface.txt:1]: This also needs to be updated.  I will keep them in sync for now and will clean it up in a subsequent PR

File: benchmarks/dynamo/ci_expected_accuracy/aot_inductor_huggingface_inference.csv [hunk 6]
```diff
@@ -66,7 +58,7 @@ DistilBertForQuestionAnswering,pass,0
 
 
 
-DistillGPT2,pass,0
+DistillGPT2,pass,2
```
Comment (StrongerXi) [inline benchmarks/dynamo/ci_expected_accuracy/aot_inductor_huggingface_inference.csv:61]: @anijain2305 did we end up getting more graph breaks?

Comment (anijain2305) [reply benchmarks/dynamo/ci_expected_accuracy/aot_inductor_huggingface_inference.csv:61]: Yes, this one is hard to resolve. Current codebase is looking for substring like question answering or maskedlm to define a loss function. And this model name does not have it, so we have a silly logger warning graph break
